cmd/libsnap-confine-private/snap-test.c

```diff
@@ -519,71 +553,66 @@ static void test_sc_snap_split_instance_name_basic(void)
 				    sizeof instance);
 	g_assert_cmpstr(name, ==, "foo");
 	g_assert_cmpstr(instance, ==, "bar");
+}
 
-	memset(name, 0xff, sizeof name);
-	memset(instance, 0xff, sizeof instance);
-	sc_snap_split_instance_name("foo-bar_bar", name, sizeof name, instance,
-				    sizeof instance);
-	g_assert_cmpstr(name, ==, "foo-bar");
-	g_assert_cmpstr(instance, ==, "bar");
-
-	memset(name, 0xff, sizeof name);
-	memset(instance, 0xff, sizeof instance);
-	sc_snap_split_instance_name("foo-bar", name, sizeof name, instance,
-				    sizeof instance);
-	g_assert_cmpstr(name, ==, "foo-bar");
-	g_assert_cmpstr(instance, ==, "");
+static void test_sc_snap_split_snap_component_basic(void)
+{
+	char snap_name[SNAP_NAME_LEN + 1] = { 0xff };
+	char component_name[SNAP_NAME_LEN + 1] = { 0xff };
 
-	memset(name, 0xff, sizeof name);
-	memset(instance, 0xff, sizeof instance);
-	sc_snap_split_instance_name("_baz", name, sizeof name, instance,
-				    sizeof instance);
-	g_assert_cmpstr(name, ==, "");
-	g_assert_cmpstr(instance, ==, "baz");
+	sc_snap_split_snap_component("foo+bar", snap_name, sizeof snap_name,
+				     component_name, sizeof component_name);
+	g_assert_cmpstr(snap_name, ==, "foo");
+	g_assert_cmpstr(component_name, ==, "bar");
+}
 
-	memset(name, 0xff, sizeof name);
-	memset(instance, 0xff, sizeof instance);
-	sc_snap_split_instance_name("foo", name, sizeof name, instance,
-				    sizeof instance);
-	g_assert_cmpstr(name, ==, "foo");
-	g_assert_cmpstr(instance, ==, "");
+static void test_sc_snap_component_validate(void)
+{
+	sc_error *err = NULL;
+	sc_snap_component_validate("snapname+compname", NULL, &err);
+	g_assert_null(err);
 
-	memset(name, 0xff, sizeof name);
-	sc_snap_split_instance_name("foo_bar", name, sizeof name, NULL, 0);
-	g_assert_cmpstr(name, ==, "foo");
+	sc_snap_component_validate("snap-name+comp-name", NULL, &err);
+	g_assert_null(err);
 
-	memset(instance, 0xff, sizeof instance);
-	sc_snap_split_instance_name("foo_bar", NULL, 0, instance,
-				    sizeof instance);
-	g_assert_cmpstr(instance, ==, "bar");
+	// check that we fail if the snap name isn't in the snap component
+	sc_snap_component_validate("snapname+compname", "othername", &err);
```

Comment:
can you add a test for `snapname_foo+compname & snapname` and `snapname_foo+compname & othername` ?

Reply:
This shouldn't be a allowed by `sc_snap_component_validate`, at least for how it is used right now. I added a test to ensure that it fails on strings that look like that.